xform/src/compress.rs

```diff
@@ -56,33 +56,58 @@ fn compress_file(args: &Args, pathname: &PathBuf) -> io::Result<()> {
     out_buf.extend_from_slice(&encoder.close()?);
     let out_buf_size = out_buf.len();
 
-    if args.stdout {
-        io::stdout().write_all(&out_buf)?;
-    } else {
-        let fname = format!("{}.Z", pathname.file_name().unwrap().to_str().unwrap());
-        // If "on adding .Z" the name exceeds the limit of NAME_MAX, then we output on the stdout
-        if fname.len() > NAME_MAX {
+    // check if the output buffer size is greater than input buffer size
+    // this can indicate that the compressed file is greater than the input file
+    //
+    // in such case using '-f' works, i.e "force", which means we don't care if size
+    // isn't reduced, just compress it
+    if (out_buf_size < inp_buf_size) || args.force {
+        if args.stdout {
             io::stdout().write_all(&out_buf)?;
         } else {
-            let mut new_file = pathname.clone();
-            new_file.set_file_name(format!("{fname}"));
-            let mut f = File::create(&new_file)?;
-
-            fs::remove_file(&pathname)?;
-            f.write_all(&out_buf)?;
-
-            if args.verbose {
-                println!(
-                    "{}: -- replaced with {} Compression: {:.1}%",
-                    pathname.display(),
-                    new_file.display(),
-                    100_f32 - (out_buf_size as f32 / inp_buf_size as f32) * 100_f32
-                );
+            let fname = format!("{}.Z", pathname.file_name().unwrap().to_str().unwrap());
+            // If "on adding .Z" the name exceeds the limit of NAME_MAX, then we output on the stdout
+            if fname.len() > NAME_MAX {
+                io::stdout().write_all(&out_buf)?;
+            } else {
+                let path = Path::new(&fname);
+                if path.exists() && !args.force {
+                    print!("Do you want to overwrite {fname} (y)es or (n)o ?");
```

Comment:
For what it's worth, an example prompt elsewhere in posixutils: https://github.com/rustcoreutils/posixutils-rs/blob/main/tree/src/cp.rs#L117